fs/ggml/gguf.go

```diff
@@ -541,18 +555,68 @@ func WriteGGUF(ws io.WriteSeeker, kv KV, ts []Tensor) error {
 	})
 
 	var s uint64
+	for i := range ts {
+		if err := ggufWriteTensorInfo(ws, ts[i]); err != nil {
+			return err
+		}
+		ts[i].Offset = s + uint64(ggufPadding(int64(s), int64(alignment)))
+		s += ts[i].Size()
```

Comment:
The way it was structured before is to write padding iff there's more tensors to write

Reply:
Hmm...  
There were various bugs I ran into with round-tripping.  One was offsets were messed up in the resulting file and it wouldn't load.  This looked like a bug where t was a copy.  I'll take another pass through this code.